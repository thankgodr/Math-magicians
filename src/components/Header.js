import { Link } from 'react-router-dom';
import NAVIGATIONPATH from '../common/routepaths';

const Header = () => (
     <nav className='navbar navbar-expand-lg navbar-light bg-light justify-content-between p-2'>
        <h1 className='navbar-brand'>Magic Calculator</h1>
        <ul className='navbar-nav'>
            <li className='nav-item'><Link className='nav-link' to={NAVIGATIONPATH.HOME}>Home</Link></li>
            <li className='nav-item'><Link className='nav-link' to={NAVIGATIONPATH.CALCULATOR}>Calculator</Link></li>
            <li className='nav-item'><Link className='nav-link' to={NAVIGATIONPATH.QUOTES}>Quote</Link></li>
        </ul>
    </nav>
);

export default Header;