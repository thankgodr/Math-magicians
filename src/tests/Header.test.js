import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

describe('Testing snapshot of Header', () => {
  test('Snapshot of Header view', () => {
    const elem = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
