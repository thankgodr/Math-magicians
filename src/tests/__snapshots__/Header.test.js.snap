// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing snapshot of Header Snapshot of Header view 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light justify-content-between p-2"
>
  <h1
    className="navbar-brand"
  >
    Magic Calculator
  </h1>
  <ul
    className="navbar-nav"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/cal"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/quotes"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
